tests/test_compile.py

```diff
@@ -0,0 +1,118 @@
+import os
+import shutil
+import tempfile
+
+from jinja2 import Environment
+from jinja2 import loaders
+
+
+class TestCompileTemplate:
+    archive = None
+
+    def setup(self):
+        self.archive = tempfile.mkdtemp()
+
+    def teardown(self):
+        shutil.rmtree(self.archive)
+
+    def test_filters(self):
+        env = Environment(
+            loader=loaders.DictLoader(
+                {
+                    "foo": """
+{{ 1|filter1 }}
+{{ 2|filter2 }}
+"""
+                }
+            )
+        )
+
+        env.filters["filter1"] = lambda value: value
+        env.filters["filter2"] = lambda value: value
+
+        env.compile_templates(self.archive, zip=None, ignore_errors=False)
+
+        files = os.listdir(self.archive)
+        assert len(files) == 1
+
+        with open(os.path.join(self.archive, files[0])) as f:
+            assert f.read().splitlines() == [
+                "from __future__ import generator_stop",
```

Comment:
These tests will be very brittle, they'll need to be updated every time something unrelated changes the compiled code. Ideally, determine a way to test this without inspecting the generated code, or at least only look for a specific pattern.

Reply:
Yeah, I changed to check appearence order of some keywords.